cpg_pipes/pipeline/pipeline.py

```diff
@@ -618,25 +620,34 @@ def _outputs_are_reusable(self, expected_out: ExpectedResultT) -> bool:
         Returns outputs that can be reused for the stage for the target,
         or None of none can be reused
         """
-        if not expected_out:
-            reusable = False
-        elif not self.check_expected_outputs:
-            if self.assume_outputs_exist or self.skipped:
-                # Do not check the files' existence, trust they exist.
-                # note that for skipped stages, we automatically assume outputs exist
-                reusable = True
-            else:
-                # Do not check the files' existence, assume they don't exist:
-                reusable = False
-        else:
-            # Checking that expected output exists:
-            paths: list[Path]
-            if isinstance(expected_out, dict):
-                paths = [v for k, v in expected_out.items()]
-            else:
-                paths = [expected_out]
-            reusable = all(exists(path) for path in paths)
-        return reusable
+        if self.assume_outputs_exist:
+            return True
+
+        paths: list[Path] = []
+
+        def _find_paths(val):
+            """Recursively find every Path object"""
+            if isinstance(val, Path):
+                paths.append(val)
+            if isinstance(val, list):
+                for el in val:
+                    _find_paths(el)
+            if isinstance(val, dict):
+                for el in val.values():
+                    _find_paths(el)
+
+        _find_paths(expected_out)
+
+        if paths and get_config()['workflow'].get('check_expected_outputs'):
+            # Checking that all paths in the expected output exist:
+            return all(exists(path) for path in paths)
```

Comment:
Not blocking, but just checking in terms of design: is the plan to move away from checking the existence of files and instead rely on metamist to determine the state of everything?
(@illusional FYI)

Reply:
I didn't plan to make metamist a hard dependency, but I'm open to this suggestion!

Reply:
I'm more open to relying on metamist more, since now it's very stable and has a GUI